### PR TITLE
Reintroduce JITMs in the WC Admin.

### PR DESCRIPTION
**Pull request by AnnaMag**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Enable the JITMs in the Admin sections. 

screenshot-fermarichal jurassic tube-2022 09 09-17_17_47

screenshot-fermarichal jurassic tube-2022 09 09-17_14_11

screenshot-fermarichal jurassic tube-2022 09 09-17_36_36

The JITMs haven't been added to the Onboarding wizard in order to keep the same experience. We can add them later if we want.

Closes # 60-gh-woocommerce/mothra-private

### How to test the changes in this Pull Request:

1. Install WC on a connected Jetpack site.
2. To be able to see a testing JITM, you'll need to add `&test_jitm=/1/` at the end of a WooCommerce URL. It will work on every WooCommerce page except for: `Products > Categories`, `Products > Tags`, `Products > Attributes`, and `Products > Reviews`. 
3. Verify that the JITMs look good using the new navigation.
4. They should look good also with different browser widths.
5. Verify that they are not being shown in the Onboarding wizard.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Admin/Loader.php
Comment (beaulebens) [inline plugins/woocommerce/src/Internal/Admin/Loader.php:235]: I think (requires testing) if we remove this, then we will probably completely disable all of the admin-notices-cleanup work that WC Admin does (e.g. any notices injected by any plugin will come back on all screens). JITMs do also support being inserted into a different container (`<div id="jp-admin-notices" />`) so maybe we'd be better off keeping the existing cleanup, and then carefully placing that container?

Comment (octaedro) [reply plugins/woocommerce/src/Internal/Admin/Loader.php:235]: Hi @beaulebens, thank you for your comment,

I've been working on this PR. Since the class `woocommerce-layout__notice-list-hide` essentially only adds a `display none` (so it's not being shown), I removed the class and replaced the div id with `jp-admin-notices`, otherwise, we would add the JITMs twice and hide one of them.

I couldn't find a scenario where that change breaks the experience, but if we can find one I'll leave that div as it is now and add a new one with the id `jp-admin-notices`.

**Review by louwie17 — COMMENTED** [review]
This tested well for me and code looks good!
I will approve this after we get a consensus on this comment: https://github.com/woocommerce/woocommerce/pull/34383/files#r959890093

As these changes will show all the notices again, like this Paypal payments for example:
Screen Shot 2022-09-12 at 12 43 58 PM


Users might indeed get a pleasant surprise, as I noticed that one of my older sites had several notices that would otherwise be hidden. But shown on **WP > Dashboard**.
I wonder if we could wrap these notices in a nicer format that matches our designs better (cc @jarekmorawski, @pmcpinto  ), but that would be out of scope if we wanted to get this in by Thursday (for the code freeze).
It might be nicer to have them part of some kind of dropdown on any WooCommerce screens so it's less intrusive.

**Comment by beaulebens** [comment]
> As these changes will show all the notices again, like this Paypal payments for example

The preference here was definitely to show JITMs again, but _not_ to suddenly start showing _all_ notices (which should ideally still be collected up and tucked away, as before.

**Comment by octaedro** [comment]
Thank you @louwie17 and @beaulebens for your feedback,
I just addressed the changes you suggested.
Could you take another look at this PR?

File: plugins/woocommerce/src/Internal/Admin/Loader.php
Comment (louwie17) [reply plugins/woocommerce/src/Internal/Admin/Loader.php:235]: Thanks for adding this back in, this is testing very well now, I am only seeing the Jetpack notices.

File: plugins/woocommerce-admin/client/stylesheets/shared/_global.scss
Comment (louwie17) [inline plugins/woocommerce-admin/client/stylesheets/shared/_global.scss]: The margin-bottom `2.5rem` feels like a lot on the older pages like products list, could we potentially also make this `1.25rem`?
Screen Shot 2022-09-12 at 5 05 47 PM

**Review by louwie17 — COMMENTED** [review]
Nice work @octaedro, this seems to test well now, I don't see all the notices but only the JITM's.
I just left one small comment on the styling, but otherwise this is great!

**Comment by Timstreep** [comment]
@beaulebens there's a few legacy JITMs hiding in the system that should be tested before this goes live. I've documented these [here](https://docs.google.com/spreadsheets/d/1aDI6nIKl_H3yzr2SOpWWjEMzBWokltzZVFWQ-C9LTVw/edit#gid=0). They're pretty old, so there's no telling how they might look when reintroduced or if the links still work properly. I wouldn't be opposed to disabling these altogether and starting with a clean slate.

**Comment by louwie17** [comment]
> @beaulebens there's a few legacy JITMs hiding in the system that should be tested before this goes live. I've documented these [here](https://docs.google.com/spreadsheets/d/1aDI6nIKl_H3yzr2SOpWWjEMzBWokltzZVFWQ-C9LTVw/edit#gid=0). They're pretty old, so there's no telling how they might look when reintroduced or if the links still work properly. I wouldn't be opposed to disabling these altogether and starting with a clean slate.

@Timstreep I just went through that list and managed to get all of them besides one to show up. The only one I was having trouble with was the `product_addons` one. Not sure if this is related to the required of at-least one order and it not having been synced yet.

**Review by louwie17 — DISMISSED** [review]
Thanks for sticking with the several styling updates @octaedro, this is testing well for me and looking great! Thanks so much for your hard work on this.
You got my approval and is good to merge from my perspective as long as all the checks pass :)

**Review by louwie17 — APPROVED** [review]
Re-approving after rebase.

**Comment by jarekmorawski** [comment]
Hi there! Can you tell me why we're bringing the notices back to wc-admin? Most merchants find them overwhelming, and with the Inbox and the task list in place, I am not sure why we would need another notification system.

What are we hoping to achieve with this? What's our long-term goal? Do we imagine this being limited to Jetpack or open to all extensions? cc @beaulebens

**Comment by beaulebens** [comment]
They are being brought back because they provide more contextual messaging, available on each specific screen. Note that the visuals you see here are just the basic/default theme since this was originally created by Jetpack; we will be using Woo messaging/designs, and this is not available to other plugins (100% controlled by us). We can and should definitely iterate on the design to make it something that fits into WC Admin better, but right now we wanted to at least re-enable it so that we could experiment with it in different contexts.

**Comment by jarekmorawski** [comment]
I understand the value of JITMs and would love to find a better way to show them in the system. However, as demonstrated by the recent Inbox note bug (`p1663270180977999-slack-C2CM2ELTG`), people do not react positively to additional messaging in the admin UI. Maybe we could start a high-priority project where we'd focus on the notification system in general and think of ways to balance user and business needs? I'm sure we could find a middle ground and build something that would serve our needs for years to come.

**Pull request merged**
